DMCompiler/DMCompiler.cs

```diff
@@ -52,15 +52,24 @@ public static class DMCompiler {
 
             bool successfulCompile = preprocessor is not null && Compile(preprocessor.GetResult());
 
-            if (successfulCompile) {
-                Console.WriteLine($"Compilation succeeded with {WarningCount} warnings");
-
+            if (successfulCompile)
+            {
                 //Output file is the first file with the extension changed to .json
                 string outputFile = Path.ChangeExtension(settings.Files[0], "json");
                 List<DreamMapJson> maps = ConvertMaps(preprocessor.IncludedMaps);
```

Comment:
Compiling should exit early if `ErrorCount > 0` after converting the maps, and not go on to call `SaveJson()`